set.mm

```diff
@@ -145370,12 +145402,12 @@ of an open range of nonnegative integers (of length equal to the length of
        (Proof shortened by AV, 13-May-2020.) $)
     wrdnval $p |- ( ( V e. X /\ N e. NN0 )
                  -> { w e. Word V | ( # ` w ) = N } = ( V ^m ( 0 ..^ N ) ) ) $=
-      ( wcel cn0 wa cc0 cfzo co cmap cv cword chash cfv wceq cab crab wf cvv wb
-      ovexd elmapg syldan iswrdi adantl fnfzo0hash adantll jca wrdf oveq2 feq2d
-      ex syl5ibcom imp impbid1 bitrd abbi2dv df-rab syl6reqr ) CDEZBFEZGZCHBIJZ
-      KJZALZCMZEZVFNOZBPZGZAQVJAVGRVCVKAVEVCVFVEEZVDCVFSZVKVAVBVDTEVLVMUAVCHBIU
-      BCVDVFDTUCUDVCVMVKVCVMVKVCVMGVHVJVMVHVCCBVFUEUFVBVMVJVACVFBUGUHUIUMVHVJVM
-      VHHVIIJZCVFSVJVMCVFUJVJVNVDCVFVIBHIUKULUNUOUPUQURVJAVGUSUT $.
+      ( wcel cn0 wa cc0 cfzo co cmap cv cword chash cfv wceq cab wf cvv adantl
```

Comment:
It seems the proof replacement is one byte longer.  Why not keep the old one?

Reply:
Indeed, the new proof has one more step, even though metamath.c shows another count. One should fix that. @wlammen : could you easily modify metamath.c so that `MM> SHOW PROOF <label-match> / SIZE` displays, in addition to the current output, something like `There are xxx steps, including yyy essential steps` ?
Currently:
```
!SHOW PROOF wrdnval / SIZE
The proof source for "wrdnval" has 466 characters.
!SHOW PROOF wrdnvalNEW / SIZE
The proof source for "wrdnvalNEW" has 468 characters.
!SHOW PROOF wrdnval / COMPRESSED
Proof of "wrdnval":
---------Clip out the proof below this line to put it in the source file:
      ( wcel cn0 wa cc0 cfzo co cmap cv cword chash cfv wceq cab crab wf cvv wb
      ovexd elmapg syldan iswrdi adantl fnfzo0hash adantll jca wrdf oveq2 feq2d
      ex syl5ibcom imp impbid1 bitrd abbi2dv df-rab syl6reqr ) CDEZBFEZGZCHBIJZ
      KJZALZCMZEZVFNOZBPZGZAQVJAVGRVCVKAVEVCVFVEEZVDCVFSZVKVAVBVDTEVLVMUAVCHBIU
      BCVDVFDTUCUDVCVMVKVCVMVKVCVMGVHVJVMVHVCCBVFUEUFVBVMVJVACVFBUGUHUIUMVHVJVM
      VHHVIIJZCVFSVJVMCVFUJVJVNVDCVFVIBHIUKULUNUOUPUQURVJAVGUSUT $.
---------The proof of "wrdnval" (425 bytes) ends above this line.
!SHOW PROOF wrdnvalNEW / COMPRESSED
Proof of "wrdnvalNEW":
---------Clip out the proof below this line to put it in the source file:
      ( wcel cn0 wa cc0 cfzo co cmap cv cword chash cfv wceq cab wf cvv adantl
      crab wb ovexd elmapg syldan wi iswrdi fnfzo0hash jca wrdf oveq2 syl5ibcom
      ex feq2d imp impbid1 bitrd abbi2dv df-rab syl6reqr ) CDEZBFEZGZCHBIJZKJZA
      LZCMZEZVFNOZBPZGZAQVJAVGUAVCVKAVEVCVFVEEZVDCVFRZVKVAVBVDSEVLVMUBVCHBIUCCV
      DVFDSUDUEVCVMVKVBVMVKUFVAVBVMVKVBVMGVHVJVMVHVBCBVFUGTCVFBUHUIUMTVHVJVMVHH
      VIIJZCVFRVJVMCVFUJVJVNVDCVFVIBHIUKUNULUOUPUQURVJAVGUSUT $.
---------The proof of "wrdnvalNEW" (421 bytes) ends above this line.
```

Reply:
Strange, with Metamath - Version 0.198 7-Aug-2021, I get 
```
MM> sh pr wrdnvalNEW/si
The proof source for "wrdnvalNEW" has 462 characters.
```
(the other three outputs are identical with @benjub`s results). Therefore, the new proof would be shorter than the replaced one.

Counting the number of steps, I get:
* old proof: 131 steps (19 essential steps)
* new proof: 132 steps (19 essential steps)

(@benjub where/how does metamath .c shows another count?)

Reply:
Unfortunately, the antecedent `N e. NN0`cannot be removed here, because if `N e\ NN0`:

``{ w e. Word V | ( # ` w ) = N } = (/) =/= { (/) } = ( V ^m ( 0 ..^ N ) ) ) ``

(see ~mapdm0).

Reply:
> (@benjub where/how does metamath .c shows another count?)

Strange indeed. Maybe I did not rewrap the source file and that count is sensitive to it...